net/host.go

```diff
@@ -328,49 +327,36 @@ func (h *host) writeToStream(s libp2pnetwork.Stream, msg Message) error {
 	return nil
 }
 
-// readStream reads from the stream into the given buffer, returning the number of bytes read
-func readStream(stream libp2pnetwork.Stream, buf []byte) (int, error) {
-	if stream == nil {
-		return 0, errNilStream
+// readStreamMessage reads the 4-byte LE size header and message body returning the
+// message body bytes.
+func readStreamMessage(s libp2pnetwork.Stream) ([]byte, error) {
```

Comment:
I was running into bugs in this function. I chose to simplify it instead of debug it. We use a fixed, 4-byte size (instead of variable-length 16-byte size) and we allocate the buffers in here instead of passing them it=b. The majority of the callers were dynamically allocating the maximum size buffer before every call, so the new solution is most likely less resource intensive.